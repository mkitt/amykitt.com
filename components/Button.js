// @flow
import React from 'react'
import css, { before } from '../styles/css'

type Props = {
  className?: string,
}

const style = css(
  before({
    position: 'absolute',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    zIndex: 2,
    content: '""',
    backgroundColor: 'rgba(0, 0, 0, 0)',
  }),
)


const Button = ({ className, ...props }: Props) => (
  <button
    className={`${style} ${className || ''}`}
    {...props}
    type="button"
  />
)

Button.defaultProps = {
  className: '',
}

export default Button
