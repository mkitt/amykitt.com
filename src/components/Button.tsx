import styled from '@emotion/styled/macro'
import { memo } from 'react'
import useCss, { useCssType } from '../hooks/useCss'

const Button = styled.button<useCssType>(
  {
    background: 'transparent',
    border: 0,
    borderRadius: 0,
    color: 'inherit',
    cursor: 'pointer',
    font: 'inherit',
    margin: 0,
    outline: 0,
    padding: 0,
    position: 'relative',
    textAlign: 'left',
    textTransform: 'none',
    WebkitAppearance: 'button',
  },
  useCss,
)

Button.defaultProps = {
  type: 'button',
}

export default memo(Button)
