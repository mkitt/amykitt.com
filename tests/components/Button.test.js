import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../../components/Button'

const noop = () => {}

describe('Button', () => {
  let component

  beforeEach(() => {
    component = (
      <Button
        className="ButtonClassName"
        onClick={noop}
      />
    )
  })

  it('shows the correct snapshot for the Button component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
