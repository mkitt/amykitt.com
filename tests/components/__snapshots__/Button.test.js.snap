// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button shows the correct snapshot for the Button component tree 1`] = `
<button
  className="css-1fqszn3 ButtonClassName"
  onClick={[Function]}
  type="button"
/>
`;
